### PR TITLE
Add GeoLite2 ASN database

### PR DESCRIPTION
**Pull request by oalders**

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {{$NEXT}}
 
+- Added support for the GeoIP2 ASN database.
+
```
Comment (oschwald) [inline Changes]: Two newlines.

File: lib/GeoIP2/Database/Reader.pm [hunk 2]
```diff
@@ -107,6 +108,15 @@ sub _model_for_address {
         locales => $self->locales,
     );
 }
+sub asn {
```
Comment (oschwald) [inline lib/GeoIP2/Database/Reader.pm]: Newline between subs. Not sure why perltidy didn't add this.

File: lib/GeoIP2/Database/Reader.pm [hunk 3]
```diff
@@ -107,6 +108,15 @@ sub _model_for_address {
...
+    my $self = shift;
+    return $self->_model_for_address(
+        'ASN',
+        type_check => qr/^(?:GeoLite2|GeoIP2)-(?:Precision-)?ASN$/,
```
Comment (oschwald) [inline lib/GeoIP2/Database/Reader.pm]: This can just be `qr/^GeoLite2-ASN$/`

File: lib/GeoIP2/Model/ASN.pm [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    predicate => 'has_autonomous_system_number',
+);
+
+has autonomous_system_organization => (
+    is        => 'ro',
+    isa       => Str,
+    predicate => 'has_autonomous_system_organization',
+);
+
+has organization => (
```
Comment (oschwald) [inline lib/GeoIP2/Model/ASN.pm]: There isn't `organization` in GeoLite2 ASN.

File: Changes (hunk 1 above)
Comment (oalders) [inline Changes]: Fixed in 2c9e82714

File: lib/GeoIP2/Database/Reader.pm (hunk 3 above)
Comment (oalders) [inline lib/GeoIP2/Database/Reader.pm]: Fixed in 1548999457

File: lib/GeoIP2/Model/ASN.pm (hunk 4 above)
Comment (oalders) [inline lib/GeoIP2/Model/ASN.pm]: Fixed in f3775d72d0644

File: lib/GeoIP2/Database/Reader.pm (hunk 2 above)
Comment (oalders) [inline lib/GeoIP2/Database/Reader.pm]: Thanks.  This is fixed as well.

File: lib/GeoIP2/Model/ASN.pm [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          ip_address => '128.101.101.101',
+      }
+  );
+
+  print $asn->autonomous_system_number(), "\n";
+  print $asn->autonomous_system_organization(), "\n";
+
+=head1 DESCRIPTION
+
+This class provides a model for the data returned by the GeoIP2 ASN database.
```
Comment (oschwald) [inline lib/GeoIP2/Model/ASN.pm]: GeoLite2 ASN, note GeoIP2 ASN.

Comment (oalders) [inline lib/GeoIP2/Model/ASN.pm]: Fixed in 3c8b0215

File: Changes [hunk 6]
```diff
@@ -1,5 +1,8 @@
 {{$NEXT}}
 
+- Added support for the GeoIP2 ASN database.
```
Comment (oschwald) [inline Changes]: GeoLite2

Comment (oalders) [inline Changes]: Fixed in 61ff0d4b

**Pull request merged**